### ISSUE TITLE
awaiting IEnumerator struct causes an exception

### ISSUE THREAD

**Issue by kraiHD** [Cysharp/UniTask#85]
This code will cause an InvalidProgramException unless replacing "struct" to "class".

Code:
```
using System.Collections;
using Cysharp.Threading.Tasks;
using UnityEngine;

public class TestBehaviour : MonoBehaviour
{
    struct WaitWhileFirstSecond : IEnumerator
    {
        object IEnumerator.Current => null;
        bool IEnumerator.MoveNext() => Time.time < 1f;
        void IEnumerator.Reset() { }
    }

    async UniTaskVoid Start()
    {
        Debug.Log("Start");
        await new WaitWhileFirstSecond();
        Debug.Log("End");
    }
}
```

Result:
```
InvalidProgramException: Invalid IL code in TestBehaviour/<Start>d__1:MoveNext (): IL_0023: call      0x0a000014


Cysharp.Threading.Tasks.CompilerServices.AsyncUniTaskVoidMethodBuilder.Start[TStateMachine] (TStateMachine& stateMachine) (at Assets/UniTask/Runtime/CompilerServices/AsyncUniTaskVoidMethodBuilder.cs:100)
TestBehaviour.Start () (at <03154fef03c44b038c2091c47bf7828f>:0)
```

**Comment by neuecc** [comment]
Thanks, maybe it is mono runtime's bug.

```csharp
await (IEnumerator)new WaitWhileFirstSecond();
```

this workaround works fine.

I find solution so next release, will fix it.

**Comment by neuecc** [comment]
I've released 2.0.14, it fixed.
